src/mbgl/renderer/layers/render_hillshade_layer.cpp

```diff
@@ -423,78 +495,20 @@ void RenderHillshadeLayer::update(gfx::ShaderRegistry& shaders,
             }
         }
 
-        // Set up tile drawable
-        auto vertices = staticDataSharedVertices;
-        auto* indices = &staticDataIndices;
-        auto* segments = &staticDataSegments;
-
-        if (!bucket.vertices.empty() && !bucket.indices.empty() && !bucket.segments.empty()) {
-            vertices = bucket.sharedVertices;
-            indices = &bucket.indices;
-            segments = &bucket.segments;
-        }
-
-        gfx::VertexAttributeArray hillshadeVertexAttrs;
-
-        if (const auto& attr = hillshadeVertexAttrs.add(PosAttribName)) {
-            attr->setSharedRawData(vertices,
-                                   offsetof(HillshadeLayoutVertex, a1),
-                                   0,
-                                   sizeof(HillshadeLayoutVertex),
-                                   gfx::AttributeDataType::Short2);
-        }
-        if (const auto& attr = hillshadeVertexAttrs.getOrAdd(TexturePosAttribName)) {
-            attr->setSharedRawData(vertices,
-                                   offsetof(HillshadeLayoutVertex, a2),
-                                   0,
-                                   sizeof(HillshadeLayoutVertex),
-                                   gfx::AttributeDataType::Short4);
-        }
-
-        hillshadeBuilder = context.createDrawableBuilder("hillshade");
-
         if (tileLayerGroup->getDrawableCount(renderPass, tileID) > 0) {
-            tileLayerGroup->observeDrawables(renderPass, tileID, [&](gfx::Drawable& drawable) {
-                drawable.setVertexAttributes(std::move(hillshadeVertexAttrs));
-                drawable.setVertices({}, vertices->elements(), gfx::AttributeDataType::Short2);
-
-                std::vector<std::unique_ptr<gfx::Drawable::DrawSegment>> drawSegments;
-                for (std::size_t i = 0; i < segments->size(); ++i) {
-                    const auto& seg = segments->data()[i];
-                    auto segCopy = SegmentBase{
-                        // no copy constructor
-                        seg.vertexOffset,
-                        seg.indexOffset,
-                        seg.vertexLength,
-                        seg.indexLength,
-                        seg.sortKey,
-                    };
-                    drawSegments.emplace_back(hillshadeBuilder->createSegment(gfx::Triangles(), std::move(segCopy)));
-                }
-                drawable.setIndexData(indices->vector(), std::move(drawSegments));
-
-                auto imageLocation = hillshadeShader->getSamplerLocation("u_image");
-                if (imageLocation.has_value()) {
-                    drawable.setTexture(bucket.renderTarget->getTexture(), imageLocation.value());
-                }
-            });
-            continue;
+            // TODO: check whether to remove tile
+            removeTile(renderPass, tileID);
```

Comment:
You are recreating the drawable every frame. Instead of `continue` you remove the tile. You can try this fix on the shared vertices version and probably the render tests will pass. I also curios if this is actually the fix.